src/task.rs

```diff
@@ -116,13 +115,21 @@ pub unsafe extern "C" fn sys_tick() {
 //of ticks
 pub fn sleep(sleep_ticks: u64) {
     unsafe {
-        TASKS[TASK_INDEX].wake_up_tick = TICKS + sleep_ticks;
-        TASKS[TASK_INDEX].state = TaskState::Asleep;
+        TASKS[TASK_INDEX].state = TaskState::Asleep(TICKS + sleep_ticks);
```

Comment:
Just curious, do we handle the case where TICKS overflows? Doesn't this generate panic behavior in Rust?

Reply:
I looked this up. In debug mode integer overflow generates a panic, and in release mode it is defined to wrap around. So long as your board isn't running for longer than `2^64 / tick_rate` (584,942,417 years for 1000 Hz tick rate), this is fine, but I'd rather avoid behavior we know could generate an overflow.
Reference:
https://doc.rust-lang.org/stable/reference/expressions/operator-expr.html?highlight=overflow#overflow

Reply:
I think for now it's fine, and I don't really know how I would work around it. You can open a ticket for working out a way to avoid overflow though

Reply:
I think I'm going to just add an issue for it.